src/useMultiRootEditor.tsx

```diff
@@ -42,7 +42,7 @@ const useMultiRootEditor = ( props: MultiRootHookProps ): MultiRootHookReturns =
 
 	const shouldUpdateEditor = useRef<boolean>( true );
 
-	const toolbarElement = <div dangerouslySetInnerHTML={{ __html: editor ? editor.ui.view.toolbar.element!.outerHTML : '' }}></div>;
+	const [ toolbarElement, setToolbarElement ] = useState<JSX.Element>( <></> );
```

Comment:
JSX elements should not be stored in state[1].

[1] https://www.reddit.com/r/reactjs/comments/13xk7os/how_bad_is_putting_jsx_into_state/

Reply:
I know that it is an antipattern in "normal" React apps. Initially, I wanted to solve it using `useRef`. However, when integrators render the `toolbarElement` conditionally, the ref was lost. My second attempt was `dangerouslySetInnerHTML`, but it caused the problem that the internal toolbar listeners were not bound correctly. That's why I used state to keep that value since it is only changed after recreating the component and should not cause errors from the provided discussion.

Do you have a better idea to handle it?

Reply:
Yes... it was problematic in the case when the watchdog crashed and restarted the app. However, I will check it once again.